### PR TITLE
🔒 Upgrade to latest LTS 2.164.2

### PR DESCRIPTION
**Pull request by grdryn**
We never upgraded to 2.164.1 because there were no security fixes, but
now 2.164.2. See the changelog for all changes:

https://jenkins.io/changelog-stable/

### CODE REVIEW

**Comment by gabemontero** [comment]
all sounds good @grdryn but we are not ready yet to have the master and release-4.1 branches diverge yet, and this is coming in too late to include into our upcoming GA.

I'll unhold once the release mechanics progress such that we can take new content without affecting the 4.1 GA.

**Comment by gabemontero** [comment]
well, we've now got https://jenkins.io/security/advisory/2019-04-10/#SECURITY-1327 

if you were referring to this @grdryn in your opening descriptions apologies but I missed it

we are sorting out inclusion into the 4.1 GA within openshift dev but I'll minimally restart PR testing

**Comment by gabemontero** [comment]
and we are saying let's do it

File: 2/contrib/jenkins/install-jenkins-core-plugins.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ if [[ "${INSTALL_JENKINS_VIA_RPMS}" == "false" ]]; then
     if [ "$#" == "1" ]; then
         YUM_FLAGS="$1"
     fi
-    yum -y $YUM_FLAGS --setopt=tsflags=nodocs install jenkins-2.150.2-1.1
+    yum -y $YUM_FLAGS --setopt=tsflags=nodocs install jenkins-2.164.2-1.1
     rpm -V jenkins-2.150.2-1.1
```
Comment (gabemontero) [inline 2/contrib/jenkins/install-jenkins-core-plugins.sh]: @grdryn you need to update this rpm -V jenkins-2-150 line to saw 2-164

**Comment by gabemontero** [comment]
Ah ... we have a version mismatch in our dockerfile:

```
Total download size: 74 M
Installed size: 74 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : jenkins-2.164.2-1.1.noarch                                   1/1 
  Verifying  : jenkins-2.164.2-1.1.noarch                                   1/1 

Installed:
  jenkins.noarch 0:2.164.2-1.1                                                  

Complete!
package jenkins-2.150.2-1.1 is not installed
error: build error: running '/usr/local/bin/install-jenkins-core-plugins.sh /opt/openshift/base-plugins.txt &&     rmdir /var/log/jenkins &&     chmod 664 /etc/passwd &&     chmod -R 775 /etc/alternatives &&     chmod -R 775 /var/lib/alternatives &&     chmod -R 775 /usr/lib/jvm &&     chmod 775 /usr/bin &&     chmod 775 /usr/lib/jvm-exports &&     chmod 775 /usr/share/man/man1 &&     mkdir -p /var/lib/origin &&     chmod 775 /var/lib/origin &&     unlink /usr/bin/java &&     unlink /usr/bin/jjs &&     unlink /usr/bin/keytool &&     unlink /usr/bin/orbd &&     unlink /usr/bin/pack200 &&     unlink /usr/bin/policytool &&     unlink /usr/bin/rmid &&     unlink /usr/bin/rmiregistry &&     unlink /usr/bin/servertool &&     unlink /usr/bin/tnameserv &&     unlink /usr/bin/unpack200 &&     unlink /usr/lib/jvm-exports/jre &&     unlink /usr/share/man/man1/java.1.gz &&     unlink /usr/share/man/man1/jjs.1.gz &&     unlink /usr/share/man/man1/keytool.1.gz &&     unlink /usr/share/man/man1/orbd.1.gz &&     unlink /usr/share/man/man1/pack200.1.gz &&     unlink /usr/share/man/man1/policytool.1.gz &&     unlink /usr/share/man/man1/rmid.1.gz &&     unlink /usr/share/man/man1/rmiregistry.1.gz &&     unlink /usr/share/man/man1/servertool.1.gz &&     unlink /usr/share/man/man1/tnameserv.1.gz &&     unlink /usr/share/man/man1/unpack200.1.gz &&     chown -R 1001:0 /opt/openshift &&     /usr/local/bin/fix-permissions /opt/openshift &&     /usr/local/bin/fix-permissions /opt/openshift/configuration/init.groovy.d &&     /usr/local/bin/fix-permissions /var/lib/jenkins &&     /usr/local/bin/fix-permissions /var/log' failed with exit code 1

```

File: 2/contrib/jenkins/install-jenkins-core-plugins.sh (hunk 1 above)
Comment (gabemontero) [reply 2/contrib/jenkins/install-jenkins-core-plugins.sh]: or switch back to use a VAR with the version like we used to do

**Comment by grdryn** [comment]
@gabemontero Sorry about that, I missed the `rpm -V`!

Also, yeah, the reason I created the PR was because of the security advisories. I'll be more clear about that next time :)

**Comment by grdryn** [comment]
Different error than last time (seemingly a yum repo error rather than a network-y one).

**Comment by gabemontero** [comment]
/approve
/lgtm
thanks @grdryn !!

**Comment by gabemontero** [comment]
rpm update pipeline job https://buildvm.openshift.eng.bos.redhat.com:8443/job/devex/job/devex%252Fjenkins-bump-version/32/

**Comment by gabemontero** [comment]
dist git / brew rpm commit level http://pkgs.devel.redhat.com/cgit/rpms/jenkins/commit/?h=rhaos-4.0-rhel-7&id=d1ff25b94c7d485321713fbfb214f65303ec4991

**Comment by gabemontero** [comment]
hey @grdryn - I've been told that unicode chars in PR titles and possibly commits might mess up some of our downstream / RCM tooling

I'm editing some of your recent contributions, but please refrain from using them moving forward.

thanks

**Comment by jlebon** [comment]
@gabemontero When should we expect v2.164.2 on Quay?

**Comment by gabemontero** [comment]
@jlebon no idea

my team does not control the push to quay.io like we did with docker.io 

it is now part of the overall openshift release process

my guess is that with GA less than a month out it is not a priority

based on how critical it is for you, I'd suggest emailing aos-devel and asking there

**Comment by jlebon** [comment]
Gotcha, we'll stick with the previous location for now given that it's the most up to date.

**Pull request merged**